tests/test_e2e.py

```diff
@@ -1,37 +1,66 @@
-        metadata=client.V1ObjectMeta(
-            name=namespace,
-            labels={"illuminatio-e2e": namespace})))
-    utils.create_from_yaml(k8s_client,
-                           "e2e-manifests/01-deny-all-traffic-to-an-application.yml",
-                           namespace=namespace)
-
-    # ToDo add sleep or wait until all resources are up otherwise we have a race condition
-    # ToDo handle execptions
-    res = subprocess.run(["illuminatio", "run", "--runner-image=localhost:5000/illuminatio-runner:dev"],
-                         capture_output=True,
-                         timeout=60)
 
-    assert res.returncode == 0
-    # ToDo evaluate result
-    print(res.stdout)
+@pytest.fixture(autouse=True)
+def clean_cluster(kubernetes_utils):
+    yield  # below code is executed after test(s)
+    _, core_v1, _ = kubernetes_utils
+    # delete e2e namespaces created in test setup
+    e2e_namespaces = core_v1.list_namespace(label_selector="illuminatio-e2e")
+    for namespace in e2e_namespaces.items:
+        core_v1.delete_namespace(name=namespace.metadata.name)
+    # delete illuminatio resources
+    subprocess.check_output(["illuminatio", "clean"])
 
-    # Clean up
-    res = subprocess.run(["illuminatio", "clean"], capture_output=True)
-    assert res.returncode == 0
-    print(res.stdout)
 
-    # Clean up
-    corev1.delete_namespace(name=namespace)
+@pytest.mark.parametrize(
+    "e2e_test_case",
+    [
+        "01-deny-all-traffic-to-an-application"
+        ],
+    )
+@pytest.mark.e2e
+def test__e2e__clean_setup__results_are_expected(e2e_test_case, kubernetes_utils):
+    # unpack kubernetes client
+    k8s_client, core_v1, apps_v1 = kubernetes_utils
+    # get input and expected from test case name
+    input_manifest = E2E_INPUT_MANIFEST.format(e2e_test_case)
+    expected_yaml = E2E_EXPECTED_YAML.format(e2e_test_case)
+    # create resources to test with
+    utils.create_from_yaml(k8s_client,
+                           input_manifest)
+    # wait for test resources to be ready
+    wait_for_deployments_ready(e2e_test_case,
+                               api=apps_v1,
+                               core_api=core_v1)
+    # run illuminatio, with yaml output for later comparison
+    tmp_dir = tempfile.TemporaryDirectory()
+    result_file = open(f"{tmp_dir.name}/result.yaml", "w")
+    cmd = ["illuminatio", "run", "--runner-image", f"{E2E_RUNNER_IMAGE}", "-o", f"{result_file.name}"]
+    subprocess.check_output(cmd, timeout=120)
+    # load contents of result and expected
+    stream = open(f"{tmp_dir.name}/result.yaml", "r")
+    result = yaml.safe_load(stream)
+    assert result is not None, f"Could not load result from {stream.name}"
+    expected = yaml.safe_load(open(expected_yaml, "r"))
+    assert expected is not None,  f"Could not load expected from {expected_yaml}"
+    # assert that the correct cases have been generated and results match
+    assert "cases" in result
+    print(yaml.dump(result["cases"]))
```

Comment:
I assume this print statement was only for debugging?

Reply:
Yes, but it is also useful if tests fail later-on because the diff view in pytest is hard to read for dict comparisons IMO

Reply:
Put it in an `except` block